azure-pipelines.yml

```diff
@@ -0,0 +1,85 @@
+# ASP.NET
+# Build and test ASP.NET projects.
+# Add steps that publish symbols, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/apps/aspnet/build-aspnet-4
+
+trigger:
+  batch: true
+  branches:
+    include:
+    - '*'
+  tags:
+    include:
+    - 'v*'
+
+pr:
+- 'development'
+- 'master'
+
+pool:
+  vmImage: 'windows-latest'
+
+steps:
+# -------------------
+# Install Dependencies
+# -------------------
+- task: UseDotNet@2
+  displayName: Require .NET 2.1
+  inputs:
+    packageType: 'runtime'
+    version: '2.1.x'
+
+- task: UseDotNet@2
+  displayName: Require .NET 3.1 SDK for Build Process
+  inputs:
+    packageType: 'sdk'
+    version: '3.1.x'
+
+- task: PowerShell@2
+  displayName: Setup Required Dependencies
+  inputs:
+    filePath: 'setup.ps1'
+
+- task: PowerShell@2
+  displayName: Generate Version using GitVersion
+  inputs:
+    targetType: 'inline'
+    script: dotnet gitversion /output buildserver
+
+# -------------------
+# RUN PSAKE SCRIPT
+# -------------------
+
+- task: PowerShell@2
+  displayName: Build and Test
+  continueOnError: true
```

Comment:
I was doing some testing on Contoso and found an issue when unit tests are failing, it doesn't properly publish the test reports. Make some changes here:
* Remove `continueOnError`
* Under `inputs:`, add `pwsh: true` (this will run powershell core under Windows (which is better))